src/fiber.rs

```diff
@@ -79,7 +79,24 @@ impl FiberseqData {
             target_name,
             rg,
             center_position: None,
+        };
+
+        if let Some(s) = filters.filter_expression.as_ref() {
+            if !s.is_empty() {
+                let parser = parse_filter(s.as_str());
+                let rng = if parser.rng_name == "msp" {
+                    &mut fsd.msp
+                } else if parser.rng_name == "nuc" {
+                    &mut fsd.nuc
+                } else if parser.rng_name == "m6a" {
+                    &mut fsd.m6a
+                } else { // 5mC
+                    &mut fsd.cpg
+                };
+                apply_filter_to_range(&parser, rng).unwrap();
+            }
         }
+        fsd.clone()
```

Comment:
I would prefer to have all this logic in a function in your expression filter module. So this would all be replaced with one function call that passes a mutable reference to fsd and a reference to the expression filter object.

Reply:
[50f1012](https://github.com/fiberseq/fibertools-rs/commit/50f101278debedaef50ef82dc39db1d72c565734)